### ISSUE TITLE
Querying for modalities does not filter for sessions or subjects

### ISSUE THREAD

**Issue by Remi-Gau** [bids-standard/bids-matlab#65]
Example from the 7T-trt dataset from bids-example

```
$ tree -d sub-0[12]
sub-01
├── ses-1
│   ├── anat
│   ├── fmap
│   └── func
└── ses-2
    ├── fmap
    └── func
sub-02
├── ses-1
│   ├── anat
│   ├── fmap
│   └── func
└── ses-2
    ├── fmap
    └── func
```

### Actual behavior
```
BIDS = bids.layout(dataset_path);
bids.query(BIDS,  'modalities',  'sub',  '01',  'ses',  '2')

ans =

  1×3 cell array

    'anat'    'func'    'fmap'
```

### Expected behavior
```
BIDS = bids.layout(dataset_path);
bids.query(BIDS,  'modalities',  'sub',  '01',  'ses',  '2')

ans =

  1×2 cell array

   'func'    'fmap'
```

### Notes
This also applies to `spm_bids`

**Comment by gllmflndn** [comment]
Confirmed. It is similar to #12: `subjects`, `sessions` and `modalities` were dealt with differently than the other keywords - they are qualitatively different but it seems that the code path can be the same. `modalities` is the [last one](https://github.com/bids-standard/bids-matlab/blob/master/%2Bbids/query.m#L31) treated separately.
I will open a pull request with a potential fix.

**Comment by Remi-Gau** [comment]
Fixed by PR #71
